src/main/java/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,47 @@
+package bookstore.controller;
+
+import bookstore.dto.order.OrderDto;
+import bookstore.dto.order.OrderItemDto;
+import bookstore.service.order.OrderService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/orders")
+@RequiredArgsConstructor
+public class OrderController {
+    private OrderService orderService;
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping("/{id}")
+    public OrderDto updateStatus(@PathVariable Long id, @RequestBody String status) {
+        return orderService.setOrderStatus(id, status);
+    }
+
+    @PostMapping
+    public OrderDto placeAnOrder(@RequestBody String shippingAddress,
+                                 Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        return orderService.saveOrder(userDetails.getUsername(), shippingAddress);
+    }
+
+    @GetMapping("/get-all")
```

Comment:
Check REST rules for endpoints

Reply:
![image](https://github.com/TemkaHillfinger/book-store/assets/164332253/1a1b378d-44f7-4ae9-8b43-ca3d998da1e3)
```suggestion
    @GetMapping
```